test/test-mongocrypt-ctx-encrypt.c

```diff
@@ -2163,6 +2163,373 @@ _test_encrypt_fle2_unindexed_encrypted_payload (_mongocrypt_tester_t *tester)
+      _test_rng_data_source source = {
+         .buf = {.data = (uint8_t *) RNG_DATA, .len = sizeof (RNG_DATA)}};
+#undef RNG_DATA
+      mongocrypt_t *crypt = _crypt_with_rng (&source);
+      mongocrypt_ctx_t *ctx = mongocrypt_ctx_new (crypt);
+
+      ASSERT_OK (
+         mongocrypt_ctx_setopt_index_type (ctx, MONGOCRYPT_INDEX_TYPE_EQUALITY),
+         ctx);
+      ASSERT_OK (mongocrypt_ctx_setopt_key_id (
+                    ctx, _mongocrypt_buffer_as_binary (&user_key_id)),
+                 ctx);
+      ASSERT_OK (mongocrypt_ctx_setopt_index_key_id (
+                    ctx, _mongocrypt_buffer_as_binary (&index_key_id)),
+                 ctx);
+      ASSERT_OK (mongocrypt_ctx_explicit_encrypt_init (
+                    ctx, TEST_BSON ("{'v': 'value123'}")),
+                 ctx);
+
+      ASSERT_STATE_EQUAL (mongocrypt_ctx_state (ctx),
+                          MONGOCRYPT_CTX_NEED_MONGO_KEYS);
+      {
+         ASSERT_OK (mongocrypt_ctx_mongo_feed (
+                       ctx, TEST_FILE ("./test/data/keys/"
+                                       "12345678123498761234123456789012-local-"
+                                       "document.json")),
+                    ctx);
+         ASSERT_OK (mongocrypt_ctx_mongo_feed (
+                       ctx, TEST_FILE ("./test/data/keys/"
+                                       "ABCDEFAB123498761234123456789012-local-"
+                                       "document.json")),
+                    ctx);
+         ASSERT_OK (mongocrypt_ctx_mongo_done (ctx), ctx);
+      }
+
+      ASSERT_STATE_EQUAL (mongocrypt_ctx_state (ctx), MONGOCRYPT_CTX_READY);
+      {
+         mongocrypt_binary_t *got = mongocrypt_binary_new ();
+
+         ASSERT_OK (mongocrypt_ctx_finalize (ctx, got), ctx);
+         ASSERT_MONGOCRYPT_BINARY_EQUAL_BSON (
+            TEST_BSON ("{'v': { '$binary': { 'base64': "
+                       "'BHEBAAAFZAAgAAAAAHb62aV7+mqmaGcotPLdG3KP7S8diFwWMLM/"
+                       "5rYtqLrEBXMAIAAAAAAVJ6OWHRv3OtCozHpt3ZzfBhaxZirLv3B+"
+                       "G8PuaaO4EgVjACAAAAAAsZXWOWA+UiCBbrJNB6bHflB/"
+                       "cn7pWSvwWN2jw4FPeIUFcABQAAAAAMdD1nV2nqeI1eXEQNskDflCy8I"
+                       "7/HvvqDKJ6XxjhrPQWdLqjz+8GosGUsB7A8ee/uG9/"
+                       "guENuL25XD+"
+                       "Fxxkv1LLXtavHOlLF7iW0u9yabqqBXUAEAAAAAQSNFZ4EjSYdhI0EjR"
+                       "WeJASEHQAAgAAAAV2AE0AAAAAq83vqxI0mHYSNBI0VniQEkzZZBBDge"
+                       "Zh+h+gXEmOrSFtVvkUcnHWj/rfPW7iJ0G3UJ8zpuBmUM/"
+                       "VjOMJCY4+eDqdTiPIwX+/vNXegc8FZQAgAAAAAOuac/"
+                       "eRLYakKX6B0vZ1r3QodOQFfjqJD+xlGiPu4/PsAA==', "
+                       "'subType': '06' } }}"),
+            got);
+         mongocrypt_binary_destroy (got);
+      }
+
+      mongocrypt_ctx_destroy (ctx);
+      mongocrypt_destroy (crypt);
+   }
+   /* Test Indexed with non-zero ContentionFactor. */
+   {
```

Comment:
```suggestion
   }

   /* Test Indexed with non-zero ContentionFactor. */
   {
```
Consistent spacing between test cases.